From 2bff25dd35d8b3cb90809d2798468188e0b3df5d Mon Sep 17 00:00:00 2001
From: Kruti Pendharkar <kp025370@broadcom.com>
Date: Tue, 18 Feb 2025 01:13:55 -0800
Subject: [PATCH] Fix function pointer definition mismatch

Caught with -std=c23 option which is the default for gcc-15

Pull request: https://github.com/vmware/open-vm-tools/pull/751
Addresses issue: https://github.com/vmware/open-vm-tools/issues/750
---
 open-vm-tools/AUTHORS       | 3 +++
 open-vm-tools/lib/lock/ul.c | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/open-vm-tools/AUTHORS b/open-vm-tools/AUTHORS
index b871a0235..0a6587dcb 100644
--- a/open-vm-tools/AUTHORS
+++ b/open-vm-tools/AUTHORS
@@ -101,3 +101,6 @@ Joseph Allen    Updated NetworkManager calls in suspend/resume scripts
 Brennan Kinney  Revise settings for vmware-user.desktop
                 - https://github.com/vmware/open-vm-tools/pull/668
 
+Rudy Heitbaum  Fix build when compiling with -std=c23
+                - https://github.com/vmware/open-vm-tools/pull/751
+
diff --git a/open-vm-tools/lib/lock/ul.c b/open-vm-tools/lib/lock/ul.c
index d376a98af..87ffc8dc6 100644
--- a/open-vm-tools/lib/lock/ul.c
+++ b/open-vm-tools/lib/lock/ul.c
@@ -29,7 +29,7 @@
 static Bool mxInPanic = FALSE;  // track when involved in a panic
 static Bool mxUserCollectLockingTree = FALSE;
 
-Bool (*MXUserTryAcquireForceFail)() = NULL;
+Bool (*MXUserTryAcquireForceFail)(const char *name) = NULL;
 
 static MX_Rank (*MXUserMxCheckRank)(void) = NULL;
 static void (*MXUserMxLockLister)(void) = NULL;
